clang/test/Driver/clang-offload-bundler.c

```diff
@@ -341,6 +342,17 @@
 // RUN: diff %t.tgt1 %t.res.tgt1
 // RUN: diff %t.tgt2 %t.res.tgt2
 
+// Check archive mode.
+// RUN: clang-offload-bundler -type=a -targets=host-%itanium_abi_triple,openmp-powerpc64le-ibm-linux-gnu,openmp-x86_64-pc-linux-gnu -outputs=%t.host.a,%t.tgt1.a,%t.tgt2.a -inputs=%t.a -unbundle
+// RUN: cmp %t.host.a %t.a
```

Comment:
Just to make sure I understand, why exactly does the archive end up being the same size as the extracted host object?

Reply:
The host bundle is handled in a special way, we just copy input archive as is to the output file. Please see ClangOffloadBundle.cpp, lines 1044-1048.

Reply:
I see, thanks! What perplexes me here is that I thought the driver's archive-based flow to be sending the unbundled host objects right to the linker, which in turn would expect (?) them to be "true" C++ object files and not the "fat" ones. Is this being changed somehow, or am I misinterpreting the flow/change itself?

Reply:
No, there are no changes here, we have always been using fat objects as host objects. Fat object (non LTO) is actually a host object with a few additional sections that contain target objects as data. These sections are not expected to be included into the linked image because they are marked with a special attribute that tells linker to discard them from final image.